File: Hexes/src/marcos/games/hexes/core/Vertex.java
package marcos.games.hexes.core;

import cc.lib.game.IVector2D;
import cc.lib.utils.Reflector;

public class Vertex extends Reflector<Vertex> implements IVector2D {

	static {
		addAllFields(Vertex.class);
	}
	
	public Vertex() {
		this(0,0);
	}
	
	public Vertex(int x, int y) {
		this.x = x;
		this.y = y;
	}

	final int x,y;
	final int [] p = new int[6];
	private int n = 0;
	
	void addPiece(int index) {
		p[n++] = index;
	}
	
	public final int getNum() {
		return n;
	}
	
	@Override
	public float getX() {
		return x;
	}
	@Override
	public float getY() {
		return y;
	}

	@Override
	public String toString() {
		return "[" + x + "," + y + "]";
	}

	@Override
	public boolean equals(Object obj) {
		if (this== obj)
			return true;
		Vertex v = (Vertex)obj;
		return x==v.x && y==v.y;
	}

	
}

File: Hexes/src/marcos/games/hexes/core/Vertex.kt
package marcos.games.hexes.core

import cc.lib.game.IVector2D
import cc.lib.reflector.Reflector

class Vertex constructor(x: Number = 0, y: Number = 0) : Reflector<Vertex>(), IVector2D {

	override val x: Float = x.toFloat()
	override val y: Float = y.toFloat()
	val p = IntArray(6)
	var num = 0
		private set

	fun addPiece(index: Int) {
		p[num++] = index
	}

	override fun toString(): String {
		return "[$x,$y]"
	}

	override fun equals(obj: Any?): Boolean {
		if (this === obj) return true
		val v = obj as Vertex?
		return x == v!!.x && y == v.y
	}

	companion object {
		init {
			addAllFields(Vertex::class.java)
		}
	}
}
